overlord/assertstate/assertmgr.go

```diff
@@ -95,8 +96,29 @@ func doValidateSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	err = doFetch(t.State(), snapsup.UserID, func(f asserts.Fetcher) error {
-		return snapasserts.FetchSnapAssertions(f, sha3_384)
+	modelAs, err := snapstate.Model(st)
+	if err != nil {
+		return err
+	}
+
+	err = doFetch(st, snapsup.UserID, func(f asserts.Fetcher) error {
+		if err := snapasserts.FetchSnapAssertions(f, sha3_384); err != nil {
+			return err
+		}
+
+		// fetch store assertion if available
+		if modelAs.Store() != "" {
+			err := snapasserts.FetchStore(f, modelAs.Store())
+			if notFound, ok := err.(*asserts.NotFoundError); ok {
```

Comment:
This looks like we should have something like `snapasserts.IsNotFound(err)` helper (not a blocker for this PR though) which might make this easier to read but of course there is a complication with the notFound type so maybe not. Anyway, just a idle thought.

Reply:
we do have the helper (I think is even used somewhere else in the PR)  asserts.IsNotFound but indeed in this case we care about at which level in the chain we didn't find something so we need the more complicated code